src/main/scala/is/hail/methods/MaximalIndependentSet.scala

```diff
@@ -0,0 +1,89 @@
+package is.hail.methods
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.graphx._
+
+import scala.reflect.ClassTag
+
+object MaximalIndependentSet {
+
+  def apply[VD: ClassTag, ED: ClassTag](g: Graph[VD, ED], undirected: Boolean = false): Set[Long] = {
+    // Initially set each vertex to its own degree.
+    // Start a pregel run, everyone passing a (degree, ID) pair.
+    //  -On message reception, if current degree is greater than received degree, update status to to reflect this, alert neighbors
+    //  -If current degree < received degree, status does not change, don't bother sending alert.
+
+    type Message = (Int, VertexId)
+    val pairOrd = implicitly[Ordering[Message]]
+    import pairOrd.mkOrderingOps
+
+    val initialMsg = (-1, -1L)
+
+    def receiveMessage(vertexId: VertexId, value: Message, message: Message): Message = {
+      if (message == initialMsg) {
+        value
+      }
+      else if (value < message) {
+        message
+      }
+      else {
+        value
+      }
```

Comment:
Since `initialMsg` is the smallest element ever seen in this algorithm, this method can also be implemented as 
```scala
value max message
```